app/controllers/metric_configurations_controller.rb

```diff
@@ -7,51 +6,37 @@ def exists
   end
 
   def create
-    metric_configuration_params = all_params
-
     metric_snapshot = build_metric_snapshot
-    metric_configuration_params.delete('metric')
-    @metric_configuration = MetricConfiguration.new(metric_configuration_params)
-    if !metric_snapshot.id.nil? && @metric_configuration.valid_metric_snapshot_code?(metric_snapshot.code)
-      @metric_configuration.metric_snapshot = metric_snapshot
-
-      respond_to do |format|
-        if @metric_configuration.errors.empty? && @metric_configuration.save
-          format.json { render json: {metric_configuration: @metric_configuration}, status: :created}
-        else
-          format.json { render json: {errors: @metric_configuration.errors.full_messages + metric_snapshot.errors.full_messages}, status: :unprocessable_entity }
-        end
+    @metric_configuration = MetricConfiguration.new(all_params.reject { |key, _| key == 'metric' })
+    if valid_metric_snapshot_for_configuration(metric_snapshot, @metric_configuration)
+      if save_metric_configuration_with_snapshot(@metric_configuration, metric_snapshot)
+        data = { metric_configuration: @metric_configuration }
+        status = :created
+      else
+        data = { errors: @metric_configuration.errors.full_messages + metric_snapshot.errors.full_messages }
+        status = :unprocessable_entity
       end
     else
-      respond_to do |format|
-        format.json { render json: {errors: @metric_configuration.errors.full_messages + metric_snapshot.errors.full_messages}, status: :unprocessable_entity }
-      end
+      data = { errors: @metric_configuration.errors.full_messages + metric_snapshot.errors.full_messages }
```

Comment:
They're both the same line of code but may be run on different preconditions because both `valid_metric_snapshot_for_configuration` and `save_metric_configuration_with_snapshot` may change the errors ([here](https://github.com/mezuro/kalibro_configurations/blob/add_rubocop/app/models/metric_configuration.rb#L33) and [here](https://github.com/mezuro/kalibro_configurations/blob/add_rubocop/app/controllers/metric_configurations_controller.rb#L78)), so they must be collected after the changes occur;
Maybe it can be done with a larger refactor, but I was trying to avoid that on this PR, only doing enough to make rubocop happy.